File: 0801-0900/0832-Flipping an Image/0832-Flipping an Image.cpp
class Solution {
public:
    vector<vector<int>> flipAndInvertImage(vector<vector<int>>& A) {
        int n = A[0].size();
        for (auto& row : A) {
            for (int j = 0; j < (n + 1) / 2; ++j) {
                int temp = row[j] ^ 1;
                row[j] = row[n - 1 - j] ^ 1;
                row[n - 1 - j] = temp;
            }
        }
        return A;
    }
};

File: 0801-0900/0832-Flipping an Image/0832-Flipping an Image.py
class Solution:
    def flipAndInvertImage(self, A: List[List[int]]) -> List[List[int]]:
        for i in range(len(A)):
            A[i] = list(map(lambda x: 1 - x, A[i]))[::-1]
        return A

class Solution:
    def flipAndInvertImage(self, A: List[List[int]]) -> List[List[int]]:
        for row in A:
            for j in range((len(row) + 1)//2):
                row[j], row[~j] = row[~j]^1, row[j]^1
        return A
